pennylane_lightning/core/src/simulators/lightning_kokkos/gates/GateFunctorsParam.hpp

```diff
@@ -1687,4 +1483,421 @@ template <class PrecisionT, bool inverse = false> struct rotFunctor {
+        arr(i1010) = GATESUM4(0B1010);
+        arr(i1011) = GATESUM4(0B1011);
+        arr(i1100) = GATESUM4(0B1100);
+        arr(i1101) = GATESUM4(0B1101);
+        arr(i1110) = GATESUM4(0B1110);
+        arr(i1111) = GATESUM4(0B1111);
+    }
+};
+
+#define GATEENTRY5(xx, yy) xx << 5 | yy
+#define GATETERM5(xx, yy, vyy) matrix(GATEENTRY5(xx, yy)) * vyy
+#define GATESUM5(xx)                                                           \
+    GATETERM5(xx, 0B00000, v00000) + GATETERM5(xx, 0B00001, v00001) +          \
+        GATETERM5(xx, 0B00010, v00010) + GATETERM5(xx, 0B00011, v00011) +      \
+        GATETERM5(xx, 0B00100, v00100) + GATETERM5(xx, 0B00101, v00101) +      \
+        GATETERM5(xx, 0B00110, v00110) + GATETERM5(xx, 0B00111, v00111) +      \
+        GATETERM5(xx, 0B01000, v01000) + GATETERM5(xx, 0B01001, v01001) +      \
+        GATETERM5(xx, 0B01010, v01010) + GATETERM5(xx, 0B01011, v01011) +      \
+        GATETERM5(xx, 0B01100, v01100) + GATETERM5(xx, 0B01101, v01101) +      \
+        GATETERM5(xx, 0B01110, v01110) + GATETERM5(xx, 0B01111, v01111) +      \
+        GATETERM5(xx, 0B10000, v10000) + GATETERM5(xx, 0B10001, v10001) +      \
+        GATETERM5(xx, 0B10010, v10010) + GATETERM5(xx, 0B10011, v10011) +      \
+        GATETERM5(xx, 0B10100, v10100) + GATETERM5(xx, 0B10101, v10101) +      \
+        GATETERM5(xx, 0B10110, v10110) + GATETERM5(xx, 0B10111, v10111) +      \
+        GATETERM5(xx, 0B11000, v11000) + GATETERM5(xx, 0B11001, v11001) +      \
+        GATETERM5(xx, 0B11010, v11010) + GATETERM5(xx, 0B11011, v11011) +      \
+        GATETERM5(xx, 0B11100, v11100) + GATETERM5(xx, 0B11101, v11101) +      \
+        GATETERM5(xx, 0B11110, v11110) + GATETERM5(xx, 0B11111, v11111)
+template <class PrecisionT> struct apply5QubitOpFunctor {
+    using ComplexT = Kokkos::complex<PrecisionT>;
+    using KokkosComplexVector = Kokkos::View<ComplexT *>;
+    using KokkosIntVector = Kokkos::View<std::size_t *>;
+
+    KokkosComplexVector arr;
+    KokkosComplexVector matrix;
+    KokkosIntVector wires;
+    const std::size_t n_wires = 5;
+    const std::size_t dim = static_cast<std::size_t>(1U) << n_wires;
+    std::size_t num_qubits;
+
+    apply5QubitOpFunctor(KokkosComplexVector &arr_, std::size_t num_qubits_,
+                         const KokkosComplexVector &matrix_,
+                         const std::vector<std::size_t> &wires_) {
+        Kokkos::View<const size_t *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>
+            wires_host(wires_.data(), wires_.size());
+        Kokkos::resize(wires, wires_host.size());
+        Kokkos::deep_copy(wires, wires_host);
+        arr = arr_;
+        matrix = matrix_;
+        num_qubits = num_qubits_;
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const std::size_t k) const {
+        const std::size_t kdim = k * dim;
+        INDEX(i00000, 0B00000);
+        ComplexT v00000 = arr(i00000);
```

Comment:
Same here, regarding making it a single-line statement. (I will not point out all of them).

Reply:
Keep discussion [here](https://github.com/PennyLaneAI/pennylane-lightning/pull/490#discussion_r1319014974).